doc/nrf/app_dev/device_guides/nrf54l/kmu_basics.rst

```diff
@@ -0,0 +1,63 @@
+This feature is crucial not only for private keys but also for public keys, as the :ref:`KMU can directly transfer a key to the CRACEN RAM<ug_nrf54l_crypto_kmu_cracen_peripherals>`.
+Even when keys must pass through addressable RAM, the KMU significantly reduces the risk of key exposure.
+Therefore, you should use KMU for managing secrets whenever possible.
+
+Key Types
+*********
+
+Different types of keys, such as revocable and locked keys, serve distinct purposes and have unique policies associated with their use and management.
+
+Revocable keys
+==============
+
+Keys can be set as invalid for further use. 
+For these keys, the revocation policy (RPOLICY) must be marked as ``revoked``.
+For some technical solutions, a few keys of the same key type (for example, generation 0, 1, 2) might be provisioned.
+At any given time, only one generation should be active.
+If a key generation is compromised, it should be revoked to prevent its use.
+The specifics of the revocation scheme depend on the technical solution, but it is recommended to designate one generation as non-revocable (``locked``) to prevent a Denial of Service (DoS) attack on the key's availability for the solution.
+
+Locked keys
+===========
+
+Locked keys are permanently available for use, with an RPOLICY of ``locked``.
+
+Provisioning keys for the bootloader
+************************************
+
+The bootloader can use multiple key generations for image verification (up to three for nRF54L SoCs).
+To safeguard against unauthorized provisioning by attackers, you must :ref:`provision all key generations onto the device<ug_nrf54l_developing_provision_kmu>`.
+You can adjust the number of key generations used for application verification by MCUboot with the ``CONFIG_BOOT_SIGNATURE_KMU_SLOTS`` MCUboot's Kconfig option.
+
+Limitations of operating without TEE (TF-M)
+*******************************************
+
+Operating without Trusted Execution Environment (TEE) has a few limitations.
+
+Access to the KMU is restricted to secure code.
+When an SoC runs such applications, it maintains some control over the KMU.
+A provisioned key cannot be overwritten, and its content might be blocked from use at runtime (push block mechanism) by unexpected routines.
+However, applications can revoke a revocable key on the nRF54L15, nRF54L10, and nRF54L05 devices.
+
+KMU keys are vulnerable to revocation by applications running in secure mode.
+If you cannot trust the application running in the secure mode, you should provision locked keys.
+This consideration is particularly important for applications without a TEE setup.
+For this reason, you must evaluate the security aspects of your application.
+You can consider the following options:
+
+* Supporting key generations with revocable keys - The advantage of this method is that the application can revoke a key generation if the private key is compromised or lost.
```

Comment:
```suggestion
* Supporting key generations with revocable keys - The advantage of this method is that the application or bootloader can revoke a key generation if the private key is compromised or lost.
```

Reply:
True in general. I doubt in adding bootloader there as this is not implemented yet.